crates/core/src/surfnet/mod.rs

```diff
@@ -699,8 +699,23 @@ impl SurfnetSvm {
         // svm cache, fetch them from the RPC, and insert them locally
         let accounts = match &transaction.message {
             VersionedMessage::Legacy(message) => message.account_keys.clone(),
-            VersionedMessage::V0(message) => message.account_keys.clone(),
+            VersionedMessage::V0(message) => { 
+                let alts = message.address_table_lookups.clone();
+                let mut acc_keys = message.account_keys.clone();
+                  if !alts.is_empty() {
+                    let mut alt_pubkeys = 
+                    alts
+                   .iter()
+                   .map(|msg| msg.account_key)
+                   .collect::<Vec<_>>();
+                   acc_keys.append(&mut alt_pubkeys);
+                   acc_keys
+                  } else {
+                    acc_keys
+                  }
```

Comment:
I think something's off with the indentation here, could that be fixed?

Also, I think there's no need for `if !alts.is_empty()`, you can just map over the empty array and we'll have nothing to append if empty.

Maybe:
```rust
let mut alt_account_keys = message
    .address_table_lookups
    .iter()
    .map(|msg| msg.account_key.clone())
    .collect::<Vec<_>>();

let mut account_keys = message.account_keys.clone();
account_keys.append(&mut alt_account_keys);
account_keys
```

Would have the same result with a bit less logic?